blockprod/src/detail/mod.rs

```diff
@@ -13,32 +13,70 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-mod block_maker;
-pub mod builder;
-
-use std::sync::Arc;
+use std::{
+    collections::BTreeMap,
+    sync::{atomic::AtomicBool, Arc},
+};
 
-use chainstate::ChainstateHandle;
+use chainstate::{ChainstateHandle, PropertyQueryError};
+use chainstate_types::{BlockIndex, GenBlockIndex, GetAncestorError};
 use common::{
-    chain::{block::BlockCreationError, Block, ChainConfig, Destination, SignedTransaction},
+    chain::{
+        block::{
+            calculate_tx_merkle_root, calculate_witness_merkle_root, timestamp::BlockTimestamp,
+            BlockBody, BlockCreationError, BlockHeader, BlockReward, ConsensusData,
+        },
+        Block, ChainConfig, Destination, GenBlock, SignedTransaction,
+    },
+    primitives::{BlockHeight, Id},
     time_getter::TimeGetter,
 };
-use mempool::MempoolHandle;
-use tokio::sync::mpsc;
-
-use crate::{
-    detail::block_maker::BlockMaker, interface::BlockProductionInterface, BlockProductionError,
+use futures::channel::oneshot;
```

Comment:
As we're already using Tokio, `tokio::sync::oneshot` is a drop-in a 99% drop-in replacement for `futures::channel::oneshot`

Reply:
Done

Reply:
I reverted that change because it broke the tests. Feel free to change it and fix the tests that come with it.

Reply:
Cool, no problem.